Rubberduck.Parsing/Symbols/DeclarationSymbolsListener.cs

```diff
@@ -17,14 +20,47 @@ public class DeclarationSymbolsListener : VBABaseListener
         private string _currentScope;
         private Declaration _parentDeclaration;
 
-        public DeclarationSymbolsListener(VBComponentParseResult result)
-            : this(result.QualifiedName, Accessibility.Implicit, result.Component.Type)
+        private readonly IEnumerable<CommentNode> _comments;
+
+        private string FindAnnotations()
+        {
+            if (_comments == null)
+            {
+                return null;
+            }
+
+            var lastDeclarationsSectionLine = _qualifiedName.Component.CodeModule.CountOfDeclarationLines;
+            var annotations = _comments.Where(comment => comment.QualifiedSelection.Selection.EndLine <= lastDeclarationsSectionLine
+                && comment.CommentText.StartsWith("@")).ToArray();
+
+            if (annotations.Any())
+            {
+                return string.Join("\n", annotations.Select(annotation => annotation.CommentText));
+            }
+
+            return null;
+        }
+
+        private string FindAnnotations(int line)
         {
+            if (_comments == null)
+            {
+                return null;
+            }
+
+            var commentAbove = _comments.SingleOrDefault(comment => comment.QualifiedSelection.Selection.EndLine == line - 1);
+            if (commentAbove != null && commentAbove.CommentText.StartsWith("@"))
+            {
+                return commentAbove.CommentText;
+            }
+
```

Comment:
todo: figure out how to deal with `@Ignore` annotations decorating a `@TestMethod` procedure.